src/Sulu/Bundle/DocumentManagerBundle/Initializer/Initializer.php

```diff
@@ -15,25 +15,64 @@
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
+/**
+ * Content repository initializer.
+ *
+ * Can purge and initialize the content repository using any number of
+ * registered *initializers*.
+ *
+ * Initializers should be designed to be idempotent - they should not do
+ * anything to change the state of the repository beyond their initial changes:
+ * https://en.wikipedia.org/wiki/Idempotent
+ */
 class Initializer
 {
+    /**
+     * @var ContainerInterface
+     */
     private $container;
+
+    /**
+     * @var array
+     */
     private $initializerMap;
 
-    public function __construct(ContainerInterface $container, array $initializerMap = [])
+    /**
+     * @var PurgerInterface
+     */
+    private $purger;
+
+    public function __construct(ContainerInterface $container, PurgerInterface $purger, array $initializerMap = [])
     {
         $this->container = $container;
         $this->initializerMap = $initializerMap;
+        $this->purger = $purger;
     }
 
-    public function initialize(OutputInterface $output = null)
+    /**
+     * Initialize the content repository, optionally purging it before-hand.
+     *
+     * @param OutputInterface $output
+     * @param bool $purge
+     */
+    public function initialize(OutputInterface $output = null, $purge = false)
     {
         $output = $output ?: new NullOutput();
+
+        if (true === $purge) {
+            $output->write('!! Purging workspaces');
+            $this->purger->purge();
+            $output->writeln(' [OK]');
+        }
+
         arsort($this->initializerMap);
 
         foreach (array_keys($this->initializerMap) as $initializerId) {
+            $output->writeln(sprintf('<comment>%s</>', $initializerId));
             $initializer = $this->container->get($initializerId);
```

Comment:
The idea is to lazily instantiate them. 

The problem is that the command is intantiated every time you launch the
console, so if we were to inject the services then the Initializer would
be instantatiated every time, and then every single initializer class
would also be intnatiated, even though they are not needed.

As it is, only the Initializer service will be instantiated.

On Wed, Mar 09, 2016 at 04:38:14AM -0800, Daniel Rotter wrote:
>    In [1]src/Sulu/Bundle/DocumentManagerBundle/Initializer/Initializer.php:
> 
>  >          arsort($this->initializerMap);
>  >
>  >          foreach (array_keys($this->initializerMap) as $initializerId) {
>  > +            $output->writeln(sprintf('<comment>%s</>', $initializerId));
>  >              $initializer = $this->container->get($initializerId);
> 
>    Wouldn't it be better to collect these services using a CompilerPass,
>    instead of injecting the container here?
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/2076#discussion_r55511947
>    2. https://github.com/sulu-io/sulu/pull/2076/files#r55511947
>    3. https://github.com/sulu-io/sulu/pull/2076/files#r55511947